cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -97,3 +100,57 @@ bool sc_cgroup_is_v2() {
     }
     return false;
 }
+
+void sc_cgroup_mount_snapd_hierarchy(void) {
+    struct stat stat_buf;
+    struct statfs statfs_buf;
+
+    if (lstat(snapd_cgroup_dir, &stat_buf) < 0) {
+        if (errno != ENOENT) {
+            die("cannot stat %s", snapd_cgroup_dir);
+        }
+        if (statfs(cgroup_dir, &statfs_buf) < 0) {
+            die("cannot fstfs %s", cgroup_dir);
+        }
+        if (statfs_buf.f_flags & MS_RDONLY) {
+            // Remount /sys/fs/cgroup read-write if it was read-only. It's
+            // normally switched read-only after systemd is done initializing.
+            if (mount("none", cgroup_dir, "", MS_REMOUNT, NULL) < 0) {
+                die("cannot remount %s as writable", cgroup_dir);
+            }
+        }
+        if (mkdir(snapd_cgroup_dir, 0555) < 0) {
+            die("cannot mkdir %s", snapd_cgroup_dir);
+        }
+        if (chown(snapd_cgroup_dir, 0, 0) < 0) {
+            die("cannot chown %s to root.root", snapd_cgroup_dir);
+        }
+        if (statfs_buf.f_flags & MS_RDONLY) {
+            if (mount("none", cgroup_dir, "", MS_REMOUNT | MS_RDONLY, NULL) < 0) {
```

Comment:
Could use a comment:
```
// Restore read-only option on the mount if needed
if (mount(...)
```

Reply:
This entire feature was removed. I moved the mount to /run/snapd/cgroup.